tests/unit/test_counts.py

```diff
@@ -0,0 +1,241 @@
+                    "PBCCSOnlySingleFlowcell.GetRunInfo": {"Done": 1},
+                    "PBCCSOnlySingleFlowcell.MergeCCSReports": {"Done": 1},
+                    "PBCCSOnlySingleFlowcell.MergeChunks": {"Done": 1},
+                    "PBCCSOnlySingleFlowcell.ShardLongReads": {"Done": 1},
+                    "ScatterAt51_18": {"Done": 1},
+                },
+            ],
+        ],
+    )
+    def test_print_task_status_summary(
+        self, mock_data_path, metadata_name, task_summary, capsys
+    ):
+
+        with open(mock_data_path.joinpath(metadata_name), "r") as f:
+            workflow_metadata = json.load(f)
+
+        counts_command.print_task_status_summary(workflow_metadata=workflow_metadata)
+        captured = capsys.readouterr()
+        assert captured.out.rstrip() == json.dumps(
+            task_summary, indent=4, sort_keys=True
+        )
+
+    @pytest.mark.parametrize(
+        "test_shards, shard_count",
+        # The test_shards is a very minimal representation of a task call for a
+        # workflow metadata. Normally the list contains several other pieces of call
+        # info but for this test the focus is on executionStatus and shardIndex
+        [
+            [
+                # test_shard 1: having one failed shard
+                [{"executionStatus": "Failed", "shardIndex": -1}],
+                # failed_shards
+                {"Failed": 1},
+            ],
+            [
+                # test_shard 2: having more than one shard and different statuses
+                [
+                    {"executionStatus": "Done", "shardIndex": 0},
+                    {"executionStatus": "Failed", "shardIndex": 1},
+                    {"executionStatus": "Done", "shardIndex": 2},
+                    {"executionStatus": "Failed", "shardIndex": 3},
+                ],
+                # failed_shards
+                {"Done": 2, "Failed": 2},
+            ],
+        ],
+    )
+    def test_get_shard_status_count(self, test_shards, shard_count):
+        assert (
+            counts_command.get_shard_status_count(
+                shards=test_shards,
+            )
+            == shard_count
+        )
+
+    @pytest.mark.parametrize(
+        "test_shards, failed_index",
+        [
+            [
+                # test_shard 1: having one failed shard
+                [{"executionStatus": "Failed", "shardIndex": -1}],
+                # failed_shards
+                [-1],
+            ],
+            [
+                # test_shard 2: having more than one shard and different statuses
+                [
+                    {"executionStatus": "Done", "shardIndex": 0},
+                    {"executionStatus": "Failed", "shardIndex": 1},
+                    {"executionStatus": "Done", "shardIndex": 2},
+                    {"executionStatus": "Failed", "shardIndex": 3},
+                ],
+                # failed_shards
+                [1, 3],
```

Comment:
need one more test case: nothing failed

Reply:
Added a new test function for this case

Reply:
Huh? I'm not seeing it here.

Reply:
Ok, i just added a change to the original function as suggested in another comment. Now I removed the new test function and just added another testing parameter with no failed shards